src/main/java/com/microsoft/sqlserver/jdbc/SQLServerParameterMetaData.java

```diff
@@ -765,170 +772,123 @@ public int getParameterCount() throws SQLServerException {
     @Override
     public int getParameterMode(int param) throws SQLServerException {
         checkClosed();
-        try {
-            if (procMetadata == null) {
-                checkParam(param);
-                // if it is not a stored proc, the param can only be input.
+        checkParam(param);
```

Comment:
I think this checkParam should be moved to line 777. The logic is different now from before - we didn't require isValidParamProc to be true to be able to call this method before (and successfully return from the else statement), but now we do. If this was intentional, I think we should cover this case specifically as well.